changelog/@unreleased/pr-284.v2.yml

```diff
@@ -0,0 +1,5 @@
+type: fix
+fix:
+  description: Handle server shutdown before succesful start
```

Comment:
Mind updating this to be a bit more descriptive of the bug and resolution, so that it's clear what this fixes by just reading release notes. Your "Before this PR" is a good starting point since it touches on the fact that there was a race condition and the server would remain started.

Reply:
Agreed, updated!